Initialization failure on npx react-native-windows-init --overwrite

Hi,
I have problem when I run this command it can no longer execute it
**` npx react-native-windows-init --overwrite`**
because it cannot find the package of **`
react-native-windows@^0.61.0-0
`**
```
Reading application name from package.json...
Reading react-native version from node_modules...
Reading react-native version from node_modules...
Latest matching version of react-native-windows for react-native@0.61.5 is react-native-windows@0.61.14
Installing react-native-windows@^0.61.0-0...
npm ERR! code ETARGET
npm ERR! notarget No matching version found for react-native-windows@0.61.0-0.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\alessandrobartoli\AppData\Roaming\npm-cache\_logs\2020-06-22T15_45_34_650Z-debug.log
Command failed: npm install --save react-native-windows@^0.61.0-0
npm ERR! code ETARGET
npm ERR! notarget No matching version found for react-native-windows@0.61.0-0.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\alessandrobartoli\AppData\Roaming\npm-cache\_logs\2020-06-22T15_45_34_650Z-debug.log

Error: Command failed: npm install --save react-native-windows@^0.61.0-0
npm ERR! code ETARGET
npm ERR! notarget No matching version found for react-native-windows@0.61.0-0.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
```
I hope you can help me

npm version: 6.11.3
node version: v12.11.1

--- a/change/react-native-windows-init-2020-06-23-18-14-45-yarninit.json
+++ b/change/react-native-windows-init-2020-06-23-18-14-45-yarninit.json
@@ -1,0 +1,8 @@
+{
+  "type": "patch",
+  "comment": "Fix issue running react-native-windows-init with yarn",
+  "packageName": "react-native-windows-init",
+  "email": "acoates@microsoft.com",
+  "dependentChangeType": "patch",
+  "date": "2020-06-24T01:14:45.540Z"
+}

--- a/packages/react-native-windows-init/src/Cli.ts
+++ b/packages/react-native-windows-init/src/Cli.ts
@@ -246,6 +246,43 @@ async function getLatestMatchingRNWVersion(
   }
 }
 
+function installReactNativeWindows(version: string, useDevMode: boolean) {
+  console.log(
+    `Installing ${chalk.green('react-native-windows')}@${chalk.cyan(
+      version,
+    )}...`,
+  );
+
+  const cwd = process.cwd();
+  const execOptions = argv.verbose ? {stdio: 'inherit' as 'inherit'} : {};
+
+  if (useDevMode) {
+    execSync(
+      isProjectUsingYarn(cwd)
+        ? 'yarn link react-native-windows'
+        : 'npm link react-native-windows',
+      execOptions,
+    );
+  } else {
+    const pkgJsonPath = findUp.sync('package.json', {cwd});
+    let pkgJson = require(pkgJsonPath);
+    let deps = pkgJson.dependencies || {};
+    deps['react-native-windows'] = version;
+    fs.writeFileSync(pkgJsonPath, JSON.stringify(pkgJson, null, 2));
+    execSync(isProjectUsingYarn(cwd) ? 'yarn' : 'npm install', execOptions);
+  }
+
+  console.log(
+    chalk.green(
+      `react-native-windows@${chalk.cyan(
+        require(require.resolve('react-native-windows/package.json', {
+          paths: [cwd],
+        })).version,
+      )} successfully installed.`,
+    ),
+  );
+}
+
 /**
  * Check if project is using Yarn (has `yarn.lock` in the tree)
  */
@@ -354,32 +391,7 @@ You can either downgrade your version of ${chalk.green(
       }
     }
 
-    const pkgmgr = isProjectUsingYarn(process.cwd())
-      ? useDevMode
-        ? 'yarn link'
-        : 'yarn add'
-      : useDevMode
-      ? 'npm link'
-      : 'npm install --save';
-    const execOptions = argv.verbose ? {stdio: 'inherit' as 'inherit'} : {};
-    console.log(
-      `Installing ${chalk.green('react-native-windows')}@${chalk.cyan(
-        version,
-      )}...`,
-    );
-    const pkgIdentity = useDevMode
-      ? 'react-native-windows'
-      : `react-native-windows@${version}`;
-    execSync(`${pkgmgr} ${pkgIdentity}`, execOptions);
-    console.log(
-      chalk.green(
-        `react-native-windows@${chalk.cyan(
-          require(require.resolve('react-native-windows/package.json', {
-            paths: [process.cwd()],
-          })).version,
-        )} successfully installed.`,
-      ),
-    );
+    installReactNativeWindows(version, useDevMode);
 
     const generateWindows = require(reactNativeWindowsGeneratePath());
     generateWindows(process.cwd(), name, ns, {

